pytri/__init__.py

```diff
@@ -133,7 +134,66 @@ def clear(self):
         for name in store_layers:
             self.remove_layer(name)
 
-    def axes(self):
+    def _fetch_layer_file(self, fname: str) -> str:
+        """
+        Fetch a layer file from local package resources.
+
+        Arguments:
+            fname (str)
+
+        Returns:
+            str JS
+
+        """
+        _js = ""
+        path, _ = split(__file__)
+        file = join(path, "js", fname)
+        with open(file, 'r') as fh:
+            _js = fh.read().strip()
+        return _js
+
+    def add_layer(self, layer_js: str, params: dict = None, name: str = None) -> str:
+        """
+        Add a custom JS layer to the visualization.
+
+        Arguments:
+            layer_js (str): The contents of a JS file
+            params (dict): The data to pass into the layer constructor
+            name (str): Optional name for the layer
+
+        Returns:
+            str: Name, as inserted
+
+        """
+        if layer_js.startswith("http"):
+            raise NotImplementedError("Cannot accept layers over HTTP yet.")
+
+        _js = layer_js
+
+        if name is None:
+            name = str(uuid.uuid4())
+
+        if params is None:
+            params = {}
+
+        try:
+            _js_layer_name = re.match(
+                r"[\s\S]*class (\w+) extends .*Layer[\s\S]*", _js
```

Comment:
I think I'm comfortable with requiring all layers to literally end with "Layer", but it won't work if they don't.

Reply:
au contraire — this confirms that the class `Foo` extends either `Layer` or optionally `substrate.Layer` (or `foo.Layer`, etc) if the author has chosen to namespace it.